src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -85,6 +85,7 @@ public ReadOnlySequence(ReadOnlySequenceSegment<T> startSegment, int startIndex,
         {
             if (startSegment == null ||
                 endSegment == null ||
+                (startSegment != endSegment && startSegment.RunningIndex > endSegment.RunningIndex) ||
```

Comment:
@pakrym, is this check valid?

Reply:
Yes. But with this amount of checks Pipelines would soon need a new unsafe ctor.